### PR TITLE
feat: add query suggestions docs

### PR DESCRIPTION
**Pull request by bietkul**

### CODE REVIEW

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
See the TODO notes.

File: content/docs/analytics/QuerySuggestions.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+### When to use it?
+
+Query Suggestions is useful to curate search suggestions based on actual search queries that your users are making. This can be used by itself or alongside suggestions based on the product data to provide an augmented search experience.
+
+Because query suggestions roll up the search queries based on the unique occurences in a period relevant to you, the suggestions index created this way is very fast. In addition to the automated job of popularing the query suggestions index daily, you can also add external suggestions from other sources.
+
+
+## Query Suggestions Preferences
+
+// TODO: Add screenshot of Suggestions UI
```
Comment (siddharthlatest) [inline content/docs/analytics/QuerySuggestions.md]: @bietkul This should be easy. Do it after the suggestions UI changes.

File: content/docs/analytics/QuerySuggestions.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                `;
+			},
+		},
+	},
+]);
+```
+
+### Usage Example With ReactiveSearch
+
+// TODO: The example is not complete
```
Comment (siddharthlatest) [inline content/docs/analytics/QuerySuggestions.md]: @bietkul We will deal with this in a separate issue. I would recommend trying the below demo once. Does this work on partial suggestions terms? It seems that we should have search analyzers enabled on the query suggestions index.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
See the specific comments.

File: content/docs/analytics/QuerySuggestions.md (hunk 2 above)
Comment (siddharthlatest) [reply content/docs/analytics/QuerySuggestions.md]: We actually handle this. Use both `key`^3 and `key.autosuggest`^1 and use weights appropriately.

**Comment by bietkul** [comment]
@siddharthlatest Can you review & merge it.

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**